web/modules/custom/hoeringsportal_audit_log/src/Form/SettingsForm.php

```diff
@@ -0,0 +1,197 @@
+    ];
+
+    /** @var \Drupal\Core\Entity\EntityInterface[] $definitions */
+    $definitions = $this->entityTypeManager->getDefinitions();
+
+    $form['types'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Definition routes'),
+      '#tree' => TRUE,
+    ];
+
+    foreach ($definitions as $definitionId => $definition) {
+      if (in_array($definitionId, $enabledAuditIds)) {
+
+        $bundleEntityType = $definition->getBundleEntityType();
+
+        if ($bundleEntityType) {
+          $storage = $this->entityTypeManager->getStorage($bundleEntityType);
+          $types = $storage->loadMultiple();
+        }
+        else {
+          $types = [$definition];
+        }
+
+        $linkTemplates = $definition->getLinkTemplates();
+
+        if (count($types) > 0 && count($linkTemplates) > 0) {
+          $form['types'][$definitionId] = [
+            '#type' => 'fieldset',
+            '#title' => $definition->getLabel(),
+            '#tree' => TRUE,
+            '#collapsible' => TRUE,
+          ];
+
+          foreach ($types as $type) {
+            $typeId = $type->id();
+
+            $form['types'][$definitionId][$typeId] = [
+              '#type' => 'fieldset',
+              '#title' => $type instanceof EntityTypeInterface ? $type->getLabel() : $type->label(),
+              '#tree' => TRUE,
+            ];
+
+            $options = [];
+            foreach ($linkTemplates as $path) {
+              $matches = $this->routeProvider->getRoutesByPattern($path)->all();
+              if (count($matches) > 0) {
+                $routeKey = array_key_first($matches);
+                $routeValue = reset($matches)->getPath();
+                $options[$this->configHelper->escapeProviderId($routeKey)] = $routeValue;
+              }
+            }
+
+            $form['types'][$definitionId][$typeId][] = [
+              '#type' => 'checkboxes',
+              '#options' => $options,
+              '#default_value' => $this->getDefaultValues($definitionId, $typeId),
+            ];
+
+          }
+        }
+      }
+    }
+    return $form;
+  }
+
+  /**
+   * Get default values for checkboxes.
+   */
+  private function getDefaultValues($definitionId, $typeId) : array {
+    $configTypes = $this->configHelper->getConfiguration('types');
+    $defaultValues = [];
+    if (count($configTypes) > 0) {
+      $defaultValues = reset($configTypes[$definitionId][$typeId]);
+    }
+    return $defaultValues;
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @phpstan-param array<mixed, mixed> $form
+   */
+  public function submitForm(array &$form, FormStateInterface $formState): void {
+    $loggedRouteNames = $formState->getValue('logged_pages')['logged_route_names'];
+    $this->configHelper->setConfiguration('logged_route_names', $this->fromStringToArray($loggedRouteNames));
+    $types = $formState->getValue('types');
+    if ($types) {
+      $this->configHelper->setConfiguration('types', $types);
+    }
```

Comment:
You can assume that `'types'` is always set in the form state and you should save the value even if it's empty:

 ```suggestion
    $types = $formState->getValue('types');
    $this->configHelper->setConfiguration('types', $types);
```